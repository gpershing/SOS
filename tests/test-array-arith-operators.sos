int printingarray(array int arr, int length, int current) =
if current == length
    then 1
else
    print(arr[curent]); current = current + 1; printingarray(arr, length, current)

array arr = [1,2,3,4,5]
int x = printingarray(arr, 5, 0)

array arr2 = 5 + arr
x = printingarray(arr2, 5, 0)

array arr3 = arr - 2
x = printingarray(arr3, 5, 0)

array arr4 = 2 * arr
x = printingarray(arr4, 5, 0)

array arr5 = arr/2
x = printingarray(arr5, 5, 0)

