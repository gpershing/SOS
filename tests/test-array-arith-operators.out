1
2
3
4
5
6
7
8
9
10
-1
0
1
2
3
2
4
6
8
10
0
1
1
2
2
